htlcswitch/link_test.go

```diff
@@ -670,3 +671,133 @@ func TestChannelLinkSingleHopMessageOrdering(t *testing.T) {
 		t.Fatalf("unable to make the payment: %v", err)
 	}
 }
+
+// TestChannelLinkFeeUpdate tests that a channel fee update by the
+// channel link is applied correctly.
+func TestChannelLinkFeeUpdate(t *testing.T) {
```

Comment:
Agreed. I don't think this test is needed at all. Instead it should go in `channel_test.go`. Once we have an agent that regulates the proper fee level based on info sampled from  the network, _then_ we would need something like this in order to test that it operates properly.